DOCS-1070: Pt3: Update UI on several more tutorials (#1783)
viamrobotics/docs: docs/tutorials/custom/custom-base-dog.md

Enter the path (for example, `/home/fido/robotdog/run.sh`) to your module's executable file in the **Executable path** field.
Click **Save Config** at the bottom of the page.

> ![Screenshot of the Viam app Config tab with the Modules subtab open, showing my-custom-base configured.](/tutorials/custom-base-dog/module-config.png)

Reviewer: yesssssss

## Configure the components

Now that the custom base code is set up, you need to configure all your hardware components.
Navigate to the **Components** subtab of your robot's **Config** tab.

### Configure the camera

Configure the ribbon camera on the dog as a `webcam` following our [webcam documentation](/components/camera/webcam/).

Click **Save config**.

### Configure the base

> Because your custom base relies on a local module, you need to use raw JSON to configure your modular resource.

Reviewer: [No action] Not for this PR, but I have docs for adding a local module in progress that may change this. https://github.com/viamrobotics/docs/pull/1815 if you're curious. Definitely a follow-up task, perhaps for me!

Use the **Mode** selector in the upper-left corner of the **Config** tab to switch to **Raw JSON** mode.

Locate the `"components": []` portion of the config file.
At the end of your camera configuration, add a comma and then add the following base configuration:

```json
    {
      "namespace": "rdk",
      "name": "quadruped",
      "type": "base",
      "model": "viamlabs:base:robotdog",
      "attributes": {

>           "ip_address": "<HOSTNAME>.local",

Reviewer: ```suggestion
       "ip_address": "<HOSTNAME>.local",

```

Author: ```suggestion
        "ip_address": "<HOSTNAME>.local",

```

Reviewer: Yep! My space key spaced out!!

        "port": 5001
      },
      "depends_on": []
    }
```

> Edit the `ip_address` attribute to match your robot's IP address, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`).

Reviewer: Hmm, we have to be careful here, `robotdog.local`, or anything using the word `HOSTNAME`, is _not_ an IP address. I think we have to do a terrible thing:

```suggestion
Edit the `ip_address` attribute to match your robot's hostname, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`).
```

Looking [upstream](https://github.com/viam-labs/robot-dog-module/blob/main/robotdog/src/robotdog.py), yeah it looks like `ip_address` it's just a string with no validation == just a hostname.

Best we can do downstream is use the word "hostname" everywhere we can to describe it.

Author: ooh strange ok

If this doesn't work, you can instead try using the IP address of the machine where the module is running, for example, `"ip_address": "10.0.0.123"`.

If you are using a port other than `5001`, edit the `port` attribute.
`5001` is the default port for sending and receiving instructions to and from the Freenove server.

Click **Save config**.

Your raw JSON configuration should look similar to the following:

![Raw JSON mode on the Config tab, showing the components and modules sections of the robot's raw JSON config.](/tutorials/custom-base-dog/raw-json.png)

> Toggle back to **Builder** mode and make sure a card has been generated:

Reviewer: nit: is "card" the right word here? I've been using "pane" or "configuration screen". No preference, just haven't seen "card" before.

Author: ```suggestion
Toggle back to **Builder** mode and make sure a configuration panel has been generated:
```
I don't even know; I've been using "panel" as well--I don't think we've totally standardized. TBH I like card but searching the docs I see that it's only used in the navigation service config instructions, so I won't use it.

![Screenshot of the Viam app CONFIG tab with the Config subtab open, showing quadruped configured.](/tutorials/custom-base-dog/config-modular-component.png)

> If yours doesn't look like this, go back to the raw JSON and double-check your JSON formatting.

Reviewer: ```suggestion
If your configuration does not match, go back to the raw JSON and double-check your JSON formatting.

```

Author: ```suggestion
If you don't have a correctly-generated configuration panel, go back to the raw JSON and double-check your JSON formatting.
```
I don't want to say "match" since they may have slightly different attribute values. How is this?

Reviewer: Ah, this hints that we might fail to generate the config correctly on our end (like: "unplug it and plug it back in again" style), while the issue is with config entry on the part of the user (hence: "double check your work"). You're right about `match`; I guess I'm just trying not to use "yours doesn't look like this". Since the screenshot may well differ from the actual config, what about language like the following instead, but introducing the screenshot instead of following it:

> Your configuration should resemble the following:

Author: ```suggestion
If yours doesn't resemble the following, go back to the raw JSON and double-check your JSON formatting.
```
Maybe this? I don't want to lose the main point of this which is to get them to make sure their JSON punctuation and indentation is correct, since we usually guide users to fancy config but can't here.

Reviewer: Gotcha, works for me -- thank you!
